test/Agent.t.sol

```diff
@@ -182,6 +171,30 @@ contract AgentTest is Test {
         agent.execute(permit2DatasEmpty, logics, tokensReturnEmpty);
     }
 
+    function testCannotBeInvalidOffset(uint16 offset, uint256 dataLength) external {
+        // Revert if offset + 0x24 > data.length
+        // 0x24 = 0x4(selector) + 0x20(amount)
+        vm.assume(uint256(offset) + 0x24 > dataLength);
```

Comment:
Any specific reason for using `vm.assume` instead of using `bound`?

Reply:
Both options are feasible, but `assume` is preferable here because it set a simple condition, while `bound` requires setting upper and lower limits and assigns to a variable. In this case, the condition is copied from the contract directly, I think using `assume` is more understandable.